### PR TITLE
[wip] Add an option to expand missing features for explain_prediction

### PR DESCRIPTION
**Pull request by lopuhin**
In some cases it can be interesting to check what features are missing during prediction, because missing feature weight can be higher than the sum of all active features. Here ``expand_missing_features`` option is added.

### CODE REVIEW

File: eli5/xgboost.py [hunk 1]
```diff
@@ -91,7 +93,9 @@ def explain_prediction_xgboost(
         target_names=None,
         targets=None,
         feature_names=None,
-        vectorized=False):
+        vectorized=False,
+        expand_missing_features=False,
```
Comment (kmike) [inline eli5/xgboost.py]: When is it good not to expand these features, i.e. why is it False by default?

Comment (kmike) [inline eli5/xgboost.py]: I haven't tried it, so likely you're right :) my point was that if xgboost makes decision based on these features then why shouldn't we show them. Maybe some of them have large weights and showing them is more important than showing features which are not missing. There can be a lot of present features as well; we have "top" and "feature_re" arguments to limit the display.

Comment (kmike) [inline eli5/xgboost.py]: hm, we don't have feature_re for explain_prediction methods; maybe we should - it could allow to e.g. show only missing features

Comment (lopuhin) [inline eli5/xgboost.py]: Right, that's a valid point! Let me work with it a little more, I also have a fairly limited experience with it :)

Comment (lopuhin) [inline eli5/xgboost.py]: Yes, it seems that the only place where it makes sense to collapse hidden features is for text, which is a rare use-case for tree ensembles. So let's rename it to "collapse_missing_features" and disable by default?

Comment (kmike) [inline eli5/xgboost.py]: A more general implementation could be the following: 

1. support feature_re for explain_prediction;
2. add an option to show combined weights for filtered out features.

But `collapse_missing_features` also sounds fine to me; it is already implemented, and it is a bit different from feature_re if there are non-missing features with `(missing)` in their names.

Comment (lopuhin) [inline eli5/xgboost.py]: We'll also need an ability to provide an inverse of feature_re, or make it accept a function - else we won't be able to filter out missing features, right?
I like an option to show combined weights for filtered out features, it can be useful for all classifiers, not just for xgboost.

Comment (kmike) [inline eli5/xgboost.py]: Yeah, a good point. I like 'accept function' solution.

**Comment by lopuhin** [comment]
I think https://github.com/TeamHG-Memex/eli5/pull/123#discussion_r94042895 + #126 is a cleaner approach :)
